artifacttemplates/http%3A%2F%2Fopentosca.org%2Fartifacttemplates/Ubuntu-VM_OperatingSystemInterface-w1/source/src/main/java/org/opentosca/artifacttemplates/ubuntuvm/UbuntuVMOperatingSystemInterfaceEndpoint.java

```diff
@@ -0,0 +1,180 @@
+package org.opentosca.artifacttemplates.ubuntuvm;
+
+import java.io.File;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.apache.commons.io.FileUtils;
+import org.opentosca.artifacttemplates.OpenToscaHeaders;
+import org.opentosca.artifacttemplates.SoapUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.ws.context.MessageContext;
+import org.springframework.ws.server.endpoint.annotation.Endpoint;
+import org.springframework.ws.server.endpoint.annotation.PayloadRoot;
+import org.springframework.ws.server.endpoint.annotation.RequestPayload;
+
+import static org.opentosca.artifacttemplates.ubuntuvm.FileHandler.downloadFile;
+import static org.opentosca.artifacttemplates.ubuntuvm.FileHandler.getFile;
+import static org.opentosca.artifacttemplates.ubuntuvm.FileHandler.getUrl;
+
+@Endpoint
+public class UbuntuVMOperatingSystemInterfaceEndpoint {
+
+    private static final Logger LOG = LoggerFactory.getLogger(UbuntuVMOperatingSystemInterfaceEndpoint.class);
+
+    @PayloadRoot(namespace = UbuntuVMConstants.NAMESPACE_URI, localPart = "installPackageRequest")
+    public void installPackages(@RequestPayload InstallPackageRequest request, MessageContext messageContext) {
+        LOG.info("InstallPackages request received!");
+
+        OpenToscaHeaders openToscaHeaders = SoapUtil.parseHeaders(messageContext);
+        InvokeResponse response = new InvokeResponse();
+        response.setMessageID(openToscaHeaders.messageId());
+
+        boolean success = false;
+
+        VirtualMachine VM = new VirtualMachine(request.getVMIP(), request.getVMUserName(), request.getVMPrivateKey());
+        try {
+            VM.connect();
+            success = VM.installPackages(request.getPackageNames());
+        } catch (Exception e) {
+            LOG.error("Could not install packages", e);
+            response.setError("Could not install packages: " + e.getMessage());
+        } finally {
+            VM.disconnect();
+        }
+
+        if (success) {
+            LOG.info("InstallPackages request successful");
+            response.setInstallResult("1");
+        } else {
+            response.setInstallResult("0");
+        }
+        SoapUtil.sendSoapResponse(response, InvokeResponse.class, openToscaHeaders.replyTo());
+    }
+
+    @PayloadRoot(namespace = UbuntuVMConstants.NAMESPACE_URI, localPart = "transferFileRequest")
+    public void transferFile(@RequestPayload TransferFileRequest request, MessageContext messageContext) {
+        LOG.info("TransferFile request received");
+
+        OpenToscaHeaders openToscaHeaders = SoapUtil.parseHeaders(messageContext);
+        InvokeResponse response = new InvokeResponse();
+        response.setMessageID(openToscaHeaders.messageId());
+
+        VirtualMachine VM = new VirtualMachine(request.getVMIP(), request.getVMUserName(), request.getVMPrivateKey());
```

Comment:
Variable names should be camel case. In this case only lower case

Reply:
done